compose/ingress/Dockerfile

```diff
@@ -2,3 +2,17 @@ FROM mirror.gcr.io/library/nginx:latest
 
 COPY entrypoint.sh /entrypoint.sh
 COPY nginx.conf /etc/nginx/conf.d/default.conf
+
+# Modify the default Nginx configuration to allow it to run as a non-root user
+# This follows Nginx Docker Hub instructions: https://hub.docker.com/_/nginx
+# By default, Nginx writes its PID file to /var/run/nginx.pid, which is a restricted location.
+# We change it to /tmp/nginx.pid so that it can be accessed by a non-root user.
+RUN sed -i 's|pid        /var/run/nginx.pid;|pid        /tmp/nginx.pid;|' /etc/nginx/nginx.conf && \
+    sed -i '/http {/a \
+    client_body_temp_path /tmp/client_temp;\n\
+    proxy_temp_path       /tmp/proxy_temp;\n\
+    fastcgi_temp_path     /tmp/fastcgi_temp;\n\
+    uwsgi_temp_path       /tmp/uwsgi_temp;\n\
+    scgi_temp_path        /tmp/scgi_temp;' /etc/nginx/nginx.conf
+
+USER nginx
```

Comment:
```suggestion
```

Reply:
@BrennanPaciorek Is it me or I'm not sure where your suggested change is different from the above code. 😮

Reply:
I think the best fix for this is no fix, and a comment in the dockerfile. We could do some manipulation of the file permissions to make the certificates available to the user `nginx` but this might result in lack of versatility. 

If we want to lower the container's privileges further, running the containers using podman, or restricting the users capabilities via the compose file may be the best option.

Reply:
Considering the above, I would still approve if you do find a good way to resolve the permission issues on the ingress container.